python/dask_cudf/dask_cudf/io/orc.py

```diff
@@ -74,7 +88,11 @@ def read_orc(path, columns=None, storage_options=None, **kwargs):
     dsk = {}
     N = 0
     for path, n in zip(paths, nstripes_per_file):
-        for stripe in range(n):
+        for stripe in (
+            range(n)
+            if filters is None
+            else cudf.io.orc._filter_stripes(filters, path)
+        ):
```

Comment:
Just a note (**not** something that needs to go in this PR):`read_parquet` has a `split_row_groups` argument (in both dask_cudf and upstream dask.dataframe).  The user can use this argument to assign a specific number of row-groups to each dask partition (rather than requiring a 1:1 stripe-to-partition mapping).  In the future, it would be nice to have an equivalent `split_stripes` argument for `read_orc`

Reply:
Does PyArrow's `dataset` API extend to ORC? That would seem like the ideal, but not fastest, way to implement that - have something like an `ORCDataset` made of `ORCFile`s.

Reply:
My guess is that the dataset APi would only give you the option to generate stripe fragments or file fragments.  It would still be dask_cudf's job to aggregate multiple stripe fragments to form each partition.